### ISSUE TITLE
Not reporting any Crash Event in self-host sentry sever

### ISSUE THREAD

**Issue by ultimatevegance** [getsentry/sentry-cocoa#300]
_Platform:_
- [ x ]  iOS
- [ ]  tvOS
- [ ]  MacOS
- [ ]  watchOS

_Swift:_
- [x ]  Yes -> If yes, which version?
- [ ]  No
 Swift and OC 
_sentry-cocoa installed with:_
- [ X ] CocoaPods
- [ ] Carthage
- [ ] Manually

Version of sentry-cocoa:
**3.x.x**
4.1.1
---
I have following issue:
Not reporting any Crash Event in self-host sentry sever
Description goes here ...
Not reporting any Crash Event in self-hosted sentry sever;but sending event in code manually is ok , the dashboard not showing any crash events.
**Steps to reproduce:**
- Step
 Pod install Sentry Cocoa 
config in Appdelegate: with DSN  ;
add script for uploading Symbol files in Xcode ;
create a out of range crash on a click event in real device(no debugger connected)
 
**Actual result:**
- Actual
app  crashed ,but nothing happens in dashborad ,tried a lot of times. Nothing happens...
**Expected result:**
there should be crash report in dashborad
- Result

**Comment by gkiaange1** [comment]
make sure you set the `Client.shared` before you do `try client.startCrashHandler()`

**Comment by antonioromano** [comment]
Same error here on macos, I'm setting this:
`do {
            Client.shared = try Client(dsn: "https://********@sentry.io/****")
            Client.shared?.trackMemoryPressureAsEvent()
            try Client.shared?.startCrashHandler()
        } catch let error {
print("ERROR SENTRY.IO: \(error)")
        }`
on the method `applicationDidFinishLaunching(_ aNotification: Notification)`
I also set the `Principal class` from `NSApplication` to `SentryCrashExceptionApplication`.

From logs I can see that Sentry starts successfully `Sentry Started -- Version: 4.3.4`, but after the app crashes, Sentry doesn't send the issue on the next startup

**Comment by antonioromano** [comment]
Ok, I was able to figure out how to solve it: I had to archive the application and run that version, in this way the crash events were successfully sent to the dashboard!

**Comment by Herz3h** [comment]
@antonioromano After archiving the application, and running it (from xcode). It managed to send previous exceptions. But it worked only once, and refuses to work anymore..........

**Comment by fetzig** [comment]
Have you checked the logs?
‘Client.logLevel = .verbose’

**Comment by philipphofmann** [comment]
@ultimatevegance do you still need help here or can we close this issue?

**Comment by philipphofmann** [comment]
As we didn't receive new information for quite a while, we close this now.

**Comment by dilip9071** [comment]
Hi Team 
I have implimented Sentry SDk in my iOS App.but crashes are not appearing in  my Dashboard.
Here is the my code 
 SentrySDK.start { options in
            options.dsn = "https://ac02e90facf54e96acb9cca43d8c966d@o464991.ingest.sentry.io/5476421"
            options.beforeSend = { event in
                return event
            }
            options.logLevel = SentryLogLevel.verbose
        }

Here is the my log details.

2020-10-22 08:20:20.171405+0530 EFB-DMS[3954:43406] Sentry - Debug:: Header {"sdk":{"name":"sentry.cocoa","version":"6.0.4"}}
2020-10-22 08:20:20.171736+0530 EFB-DMS[3954:43406] Sentry - Debug:: Item Header {"type":"session","length":261}
2020-10-22 08:20:20.171992+0530 EFB-DMS[3954:43406] Sentry - Debug:: Event {"errors":0,"status":"abnormal","started":"2020-10-22T02:48:14Z","did":"6CDB1D2F-B0B6-4AC5-958D-007722BA415F","duration":0,"sid":"C661BF42-506B-490B-9B59-349D4FD62FDF","attrs":{"release":"Spicejet.EFBDMS.com@1.0+1.0"},"timestamp":"2020-10-22T02:48:14Z","seq":2}
2020-10-22 08:20:20.174872+0530 EFB-DMS[3954:43406] Sentry - Debug:: SDK initialized! Version: 6.0.4
2020-10-22 08:20:20.185835+0530 EFB-DMS[3954:43406] Sentry - Debug:: Sent 0 crash report(s)

Please suggest me what i am doing wrong.
Thanks

**Comment by philipphofmann** [comment]
Hi @dilip9071, 
can you please open up a new issue for your problem and fill out our issue template.
